src/__init__.py

```diff
@@ -5402,22 +5402,27 @@ def saveIncr(self):
 
     def select(self, pyliste):
         """Build sub-pdf with page numbers in the list."""
-        if self.is_closed or self.isEncrypted:
+        if self.is_closed or self.is_encrypted:
             raise ValueError("document closed or encrypted")
         if not self.is_pdf:
             raise ValueError("is no PDF")
         if not hasattr(pyliste, "__getitem__"):
             raise ValueError("sequence required")
-        if len(pyliste) == 0 or min(pyliste) not in range(len(self)) or max(pyliste) not in range(len(self)):
+
+        valid_range = range(len(self))
+        if (len(pyliste) == 0
+            or min(pyliste) not in valid_range
+            or max(pyliste) not in valid_range
+           ):
             raise ValueError("bad page number(s)")
-        # preparatory stuff:
-        # (1) get underlying pdf document,
-        # (2) transform Python list into integer array
+
+        # get underlying pdf document,
         pdf = _as_pdf_document(self)
-        # call retainpages (code copy of fz_clean_file.c)
-        retainpages(pdf, pyliste)
-        if pdf.m_internal.rev_page_map:
-            mupdf.ll_pdf_drop_page_tree(pdf.m_internal)
+
+        # create page sub-pdf via extra.rearrange_pages2
+        extra.rearrange_pages2(pdf, tuple(pyliste))
```

Comment:
We could do:
```
if mupdf_version_tuple >= (1, 24):
    mupdf.pdf_rearrange_pages2(pdf, pyliste)
else:
    extra.rearrange_pages2(pdf, tuple(pyliste))
```
so that we check use of `mupdf.pdf_rearrange_pages2()` when testing with mupdf master. And also it makes it clear that we can remove extra.rearrange_pages2() when we move to a mupdf that provides `mupdf.pdf_rearrange_pages2()`.